stand/libsa/strcasecmp.c

```diff
@@ -41,29 +41,37 @@ static char sccsid[] = "@(#)strcasecmp.c	8.1 (Berkeley) 6/4/93";
 int
 strcasecmp(const char *s1, const char *s2)
 {
-	const u_char
-			*us1 = (const u_char *)s1,
-			*us2 = (const u_char *)s2;
+	const u_char *us1 = (const u_char *)s1,
+	    *us2 = (const u_char *)s2;
 
-	while (tolower(*us1) == tolower(*us2++))
-		if (*us1++ == '\0')
+	u_char u1, u2;
+
+	while ((u1 = tolower(*us1)) == (u2 = tolower(*us2))) {
+		if (u1 == '\0') {
 			return (0);
-	return (tolower(*us1) - tolower(*--us2));
+		}
+		us1++, us2++;
+	}
+	return (u1 - u2);
 }
 
 int
 strncasecmp(const char *s1, const char *s2, size_t n)
 {
 	if (n != 0) {
-		const u_char
-				*us1 = (const u_char *)s1,
-				*us2 = (const u_char *)s2;
+		const u_char *us1 = (const u_char *)s1,
+		    *us2 = (const u_char *)s2;
+
+		u_char u1, u2;
 
 		do {
-			if (tolower(*us1) != tolower(*us2++))
-				return (tolower(*us1) - tolower(*--us2));
-			if (*us1++ == '\0')
+			if ((u1 = tolower(*us1)) != (u2 = tolower(*us2))) {
+				return (u1 - u2);
+			}
+			if (u1 == '\0') {
 				break;
+			}
+			us1++, us2++;
```

Comment:
same comments about style and this being a larger change than just working around old compilers.